pkg/agent/apiserver/handlers/fqdncache/handler_test.go

```diff
@@ -0,0 +1,147 @@
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+
+	"antrea.io/antrea/pkg/agent/apis"
+	"antrea.io/antrea/pkg/agent/types"
+	"antrea.io/antrea/pkg/querier"
+	queriertest "antrea.io/antrea/pkg/querier/testing"
+)
+
+func TestFqdnCacheQuery(t *testing.T) {
+	expirationTime := time.Now().Add(1 * time.Hour).UTC()
+	tests := []struct {
+		name                 string
+		expectedStatus       int
+		filteredCacheEntries []types.DnsCacheEntry
+		expectedResponse     []apis.FQDNCacheResponse
+	}{
+		{
+			name:           "FQDN cache exists - multiple addresses multiple domains",
+			expectedStatus: http.StatusOK,
+			filteredCacheEntries: []types.DnsCacheEntry{
+				{
+					FQDNName:       "example.com",
+					IPAddress:      net.ParseIP("10.0.0.1"),
+					ExpirationTime: expirationTime,
+				},
+				{
+					FQDNName:       "foo.com",
+					IPAddress:      net.ParseIP("10.0.0.4"),
+					ExpirationTime: expirationTime,
+				},
+				{
+					FQDNName:       "bar.com",
+					IPAddress:      net.ParseIP("10.0.0.5"),
+					ExpirationTime: expirationTime,
+				},
+			},
+			expectedResponse: []apis.FQDNCacheResponse{
+				{
+					FQDNName:       "example.com",
+					IPAddress:      "10.0.0.1",
+					ExpirationTime: expirationTime,
+				},
+				{
+					FQDNName:       "foo.com",
+					IPAddress:      "10.0.0.4",
+					ExpirationTime: expirationTime,
+				},
+				{
+					FQDNName:       "bar.com",
+					IPAddress:      "10.0.0.5",
+					ExpirationTime: expirationTime,
+				},
+			},
+		},
+		{
+			name:                 "FQDN cache does not exist",
+			expectedStatus:       http.StatusOK,
+			filteredCacheEntries: []types.DnsCacheEntry{},
+			expectedResponse:     []apis.FQDNCacheResponse{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			q := queriertest.NewMockAgentNetworkPolicyInfoQuerier(ctrl)
+			q.EXPECT().GetFQDNCache(nil).Return(tt.filteredCacheEntries)
+			handler := HandleFunc(q)
+			req, err := http.NewRequest(http.MethodGet, "", nil)
+			require.NoError(t, err)
+			recorder := httptest.NewRecorder()
+			handler.ServeHTTP(recorder, req)
+			assert.Equal(t, tt.expectedStatus, recorder.Code)
+			var receivedResponse []apis.FQDNCacheResponse
+			err = json.Unmarshal(recorder.Body.Bytes(), &receivedResponse)
+			require.NoError(t, err)
+			assert.Equal(t, tt.expectedResponse, receivedResponse)
+		})
+	}
+}
+
+func TestNewFilterFromURLQuery(t *testing.T) {
+	samplePattern, _ := regexp.Compile("^" + strings.ReplaceAll(regexp.QuoteMeta("^example\\.com$"), `\*`, ".*") + "$")
+	tests := []struct {
+		name           string
+		queryParams    url.Values
+		expectedFilter *querier.FQDNCacheFilter
+		expectedStatus int
+	}{
+		{
+			name:           "Empty query",
+			queryParams:    url.Values{},
+			expectedFilter: nil,
+			expectedStatus: http.StatusOK, // No HTTP error
+		},
+		{
+			name: "Valid regex domain",
+			queryParams: url.Values{
+				"domain": {"^example\\.com$"},
```

Comment:
this doesn't match what a user would provide I think?
I believe you expect users to provide `example.com` and you will automatically transform it into `^example\.com$` or something like that?